src/cmd/serve.rs

```diff
@@ -763,55 +810,153 @@ pub fn serve(
+                                site = s;
                             }
-                        };
-                        messages::report_elapsed_time(start);
-                    }
-                    _ => {}
+                        }
+                        ChangeKind::Config => {
+                            // No need to iterate over change group since we're rebuilding the site.
+                            console::info("-> Config changed. The browser needs to be refreshed to make the changes visible.");
+
+                            if let Some(s) = recreate_site() {
+                                site = s;
+                            }
+                        }
+                    };
+                    messages::report_elapsed_time(start);
                 }
             }
             Err(e) => console::error(&format!("Watch error: {:?}", e)),
         };
     }
 }
 
+/// Filter `notify_debouncer_full` events. For events that we care about,
+/// return our internal simplified representation. For events we don't care about,
+/// return `None`.
+fn get_relevant_event_kind(event_kind: &EventKind) -> Option<SimpleFSEventKind> {
+    match event_kind {
+        EventKind::Create(create_kind) => match create_kind {
+            CreateKind::File | CreateKind::Folder => Some(SimpleFSEventKind::Create),
+            _ => None,
+        },
+        EventKind::Modify(modify_kind) => {
+            match modify_kind {
+                ModifyKind::Data(data_change) => match data_change {
+                    DataChange::Size | DataChange::Content => Some(SimpleFSEventKind::Modify),
+                    _ => None,
+                },
+                ModifyKind::Metadata(metadata_kind) => {
+                    // IntelliJ can modify file metadata; account for that here.
+                    // https://github.com/passcod/notify/issues/150#issuecomment-494912080
+                    match metadata_kind {
+                        MetadataKind::WriteTime
+                        | MetadataKind::Permissions
+                        | MetadataKind::Ownership => Some(SimpleFSEventKind::Modify),
+                        _ => None,
+                    }
+                }
+                ModifyKind::Name(rename_mode) => match rename_mode {
+                    RenameMode::To => Some(SimpleFSEventKind::Modify),
+                    _ => None,
+                },
+                _ => None,
+            }
+        }
+        EventKind::Remove(remove_kind) => match remove_kind {
+            RemoveKind::File | RemoveKind::Folder => Some(SimpleFSEventKind::Remove),
+            _ => None,
+        },
+        _ => None,
+    }
+}
+
+/// Return a new vector with no duplicate `PathBuf` elements. For any duplicate, keep the `PathBuf`
+/// with the latest `Instant` value.
+fn debounce_events(
+    events: &Vec<(PathBuf, SimpleFSEventKind, Instant)>,
+) -> Vec<(PathBuf, SimpleFSEventKind)> {
```

Comment:
Assuming the events are ordered (or that can will order them), we could just use a hashmap with the path as the key and insert them in order